custom_components/tahoma/climate_devices/somfy_heating_temperature_interface.py

```diff
@@ -0,0 +1,168 @@
+    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    _attr_supported_features = SUPPORT_PRESET_MODE
+
+    def __init__(self, device_url: str, coordinator: OverkizDataUpdateCoordinator):
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(2)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation i.e. heat, cool mode."""
+        if (
+            self.executor.select_state(OverkizState.CORE_ON_OFF)
+            == OverkizCommandParam.OFF
+        ):
+            return HVAC_MODE_OFF
+
+        state = self.executor.select_state(
+            OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_ACTIVE_MODE
+        )
+        if mode := OVERKIZ_TO_HVAC_MODES.get(state):
+            return mode
+
+        if state is not None:
+            # Unknown and potentially a new state, log to make it easier to report
+            _LOGGER.warning(
+                "Overkiz %s state unknown: %s",
+                OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_ACTIVE_MODE,
+                state,
+            )
+        return HVAC_MODE_OFF
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_ACTIVE_MODE, HVAC_MODES_TO_OVERKIZ[hvac_mode]
+        )
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return OVERKIZ_TO_PRESET_MODES[
+            self.executor.select_state(
+                OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_SETPOINT_MODE
+            )
+        ]
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_MANU_AND_SET_POINT_MODES,
+            PRESET_MODES_TO_OVERKIZ[preset_mode],
+        )
+
+    @property
+    def hvac_action(self) -> Optional[str]:
+        """Return the current running hvac operation if supported."""
+        current_operation = self.executor.select_state(
+            OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_OPERATING_MODE
+        )
+
+        if action := OVERKIZ_TO_HVAC_ACTION.get(current_operation):
+            return action
+
+        if current_operation is not None:
+            # Unknown and potentially a new state, log to make it easier to report
+            _LOGGER.error(
+                "Overkiz %s state unknown: %s",
+                OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_OPERATING_MODE,
+                current_operation,
+            )
+        return None
+
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Return the target temperature."""
+        if self.preset_mode not in PRESET_MODES_TO_OVERKIZ:
```

Comment:
Perhaps add some comments in this method? This one is a bit unclear if you don't own the device and don't understand why target temperature is different / why target temperature can be empty.

Reply:
I think it can also be removed. It's more logical to configure through the external number than target_temperature.

Using this function, reconfigure the current mode, even in auto mode, which can be surprising. I need to see if I can set the override temperature in auto mode.

Reply:
Done